liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/CreateServerMojo.java

```diff
@@ -174,4 +175,39 @@ private File getFileFromConfigDirectory(String file, File def) {
         } 
         return null;
     }
+    
+    /*
+     * return the filename of the project artifact to be installed by install-apps goal
+     */
+    private String getApplicationFilename() {
+        // project artifact has not be created yet when create-server goal is called in pre-package phase
+        String name = project.getBuild().getFinalName();
+        if (stripVersion) {
+            int versionBeginIndex = project.getBuild().getFinalName().lastIndexOf("-" + project.getVersion());
+            if ( versionBeginIndex != -1) {
+                name = project.getBuild().getFinalName().substring(0, versionBeginIndex);
+            }
+        }
+        
```

Comment:
Our plugin does assume the standard default configuration, since Maven is about convention over configuration.  It is always so tempted for users to override the default configuration.  I have seen many cases to use </finalName> in the pom to strip the version from the output file name.   In the loose-config issue, I am going to refactoring the installAppsMojo and installAppMojoSupport.  I can make a common method to retrieve project artifact name using a common implementation, (either using artifactId or finalName).